comp/otelcol/converter/impl/converter_test.go

```diff
@@ -147,7 +147,7 @@ func TestConvert(t *testing.T) {
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
-			converter, err := NewConverter(Requires{})
+			converter, err := NewConverterForAgent(Requires{})
 			assert.NoError(t, err)
```

Comment:
Can you please add unit test ? whats the behavior on the converter when we don't send config component ?

Reply:
only thing that changes is it skips looking for the api key in the core agent config:
https://github.com/DataDog/datadog-agent/blob/6fbbe5f680385b0550becb688dcad48a23c2b54a/comp/otelcol/converter/impl/autoconfigure.go#L156-L163
but yes will add some tests

Reply:
added another loop for TestConvert to use the ocb method, `newConverter` instead of `NewConverterForAgent`